### PR TITLE
fix string encoding when using xcom / json

### LINKED ISSUE #13535: DockerOperator / XCOM : `TypeError: Object of type bytes is not JSON serializable`
**Apache Airflow version**: 2.0.0

**Kubernetes version (if you are using kubernetes)** (use `kubectl version`): NA

**Environment**: 
* local Ubuntu 18.04 LTS / 
* docker-compose version 1.25.3, build d4d1b42b
* docker  20.10.1, build 831ebea

**What happened**:

when enabling xcom push for a docker operator the following error is thrown after the task finishes succesfully:
`TypeError: Object of type bytes is not JSON serializable`

**What you expected to happen**:

* error is not thrown
* if xcom_all is True: xcom contains all log lines
* if xcom_all is False: xcom contains last log line

**How to reproduce it**:

see docker compose and readme here:
https://github.com/AlessioM/airflow-xcom-issue

### PR DESCRIPTION
**Pull request by AlessioM**
closes: #13535

fixes the xcom issue by decoding the byte array of the log file before submitting it to xcom

### CODE REVIEW

File: airflow/providers/docker/operators/docker.py [hunk 1]
```diff
@@ -269,7 +269,12 @@ def _run_image(self) -> Optional[str]:
             # duplicated conditional logic because of expensive operation
             ret = None
             if self.do_xcom_push:
-                ret = self.cli.logs(container=self.container['Id']) if self.xcom_all else line.encode('utf-8')
+                if self.xcom_all:
+                    ret = self.cli.logs(container=self.container['Id'])
+                    if hasattr(ret, 'decode'):
+                        ret = ret.decode('utf-8')
+                else:
+                    ret = line.split('\n')[-1]
```
Comment (turbaszek) [inline airflow/providers/docker/operators/docker.py:277]: Should we always do the decode? We are planning to drop pickle for XCom and default to using JSON as serialisation mechanism.

**Comment by turbaszek** [comment]
Would you mind opening this PR to apache/airflow?

File: airflow/providers/docker/operators/docker.py (hunk 1 above)
Comment (AlessioM) [reply airflow/providers/docker/operators/docker.py:277]: this way it's consistent with the decoding before the logging (a couple of lines above) - but it also duplicates the decode call

**Comment by AlessioM** [comment]
sorry, I thought I did (not enough coffee yet):

https://github.com/apache/airflow/pull/13536

**Pull request merged**